docs/cloud-data-warehouses/snowflake.md

```diff
@@ -1 +1,414 @@
+3. View the Schema Details and Tables on this database
+4. Now change your role to TRANSFORMER_PRD -- how did this change the databases you can see?
+
+#### Set your context using SQL commands instructions
+1. Navigate to Projects
+2. Click on the + button in the top right corner -- this should open a new Snowflake Worksheet
+3. Select the RAW_DEV database -- oops! that database is no longer available because we are in the TRANSFORMER_PRD role
+4. Copy the following code into the worksheet, click the arrow next to the blue run button at the top of the screen, and select "Run All"
+    ```sql
+    USE ROLE transformer_dev;
+    USE WAREHOUSE transforming_xs_dev;
+    USE DATABASE raw_dev;
+    USE SCHEMA water_quality;
+    SHOW tables;
+    ```
+5. Notice you can also make these changes by selecting the Role and Warehouse at the top of the UI by the run button
+
+### Context is key
+As you may have noticed from the exercises above, it can be really easy to have the wrong role selected when navigating in Snowflake.
+
+Pay attention to which role and warehouse you have active, especially if you cannot find a database or table you expect to see or you receive an "object not found" error. You might just need to change your role!
+
+### Defaults
+Every user will be assigned a default Role and Warehouse.
+
+Your default ROLE should be: 
+
+- **TRANSFORMER_DEV**: This is the analytics engineer/dbt role, for transforming raw data into something analysis-ready. It has read/write/control access to both TRANSFORM and ANALYTICS, and read access to RAW.
+
+Your default WAREHOUSE should be: 
+
+- **TRANSFORMING_XS_DEV**: This warehouse is for transforming data in TRANSFORM and ANALYTICS. For most small jobs, you want XS. It gets very costly as you scale up to larger warehouses. If your job is large (it takes a while or times out to run in XS?) or it’s a long script, you can scale up, but maybe start in XS.
+
+### ODI Snowflake Architecture
+We’ve talked about Snowflake architecture generally. Now we’re going to start talking about ODI-specific architecture. The setup of our account is adapted from the approach described in this dbt blog post, which we summarize [here](https://www.getdbt.com/blog/how-we-configure-snowflake/).
+
+There are two environments set up for this project, **development** and **production**. Resources in the development environment are suffixed with DEV, and resources in the production environment are suffixed with PRD.
+
+Most of the time, developers will be working in the **development** environment. Once your branches are merged to main, they will be used in the production environment.
+
+What follows is a brief description of the most important Snowflake resources in the dev and production environments and how developers are likely to interact with them.
+
+We'll get into all of this throughout the training (and throughout our engagement with you!) -- so don't feel you need to memorize this just yet.
+
+#### **Six databases**
```

Comment:
The formatting for this section looks a weird to me, as a lot of the content is being rendered in literal boxes. Is there a way to tweak it so it's part of the main prose?